devicetypes/smartthings/zwave-siren-tamper.src/zwave-siren-tamper.groovy

```diff
@@ -0,0 +1,283 @@
+
+		log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+		Short alarmLength = (settings.alarmLength as Short) ?: 1
+		Boolean alarmLEDflash = (settings.alarmLEDflash as Boolean) ?: 0
+		Short comfortLED = (settings.comfortLED as Short) ?: 0
+		Boolean tamper = (settings.tamper as Boolean) ?: 0
+
+		if (alarmLength != state.alarmLength || alarmLEDflash != state.alarmLEDflash || comfortLED != state.comfortLED || tamper != state.tamper) {
+			state.alarmLength = alarmLength
+			state.alarmLEDflash = alarmLEDflash
+			state.comfortLED = comfortLED
+			state.tamper = tamper
+
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 1, size: 1, configurationValue: [alarmLength]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 2, size: 1, configurationValue: [alarmLEDflash ? 1 : 0]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 3, size: 1, configurationValue: [comfortLED]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 4, size: 1, configurationValue: [tamper ? 1 : 0]).format())
+			commands << "delay 1000"
+			commands << secure(zwave.basicV1.basicSet(value: 0x00))
+		}
+	}
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (state.sec) {
+			result = createEvent(descriptionText: description, displayed: false)
+		} else {
+			result = createEvent(
+				descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+				eventType: "ALERT",
+				name: "secureInclusion",
+				value: "failed",
+				displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x59: 1, 0x2B: 1, 0x71: 2, 0x85: 2, 0x25: 2, 0x7A: 1])
+	log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	log.debug "BasicReport:  $cmd"
+	[
+		createEvent([name: "switch", value: cmd.value > 0 ? "on" : "off", displayed: false]),
+		createEvent([name: "alarm", value: cmd.value > 0 ? "both" : "off"])
+	]
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	log.debug "BatteryReport:  $cmd"
+
+	def batteryLevel = cmd.batteryLevel != 255 ? cmd.batteryLevel : 0
+	def map = [name: "battery", unit: "%", value: batteryLevel]
+	createEvent(map)
+}
```

Comment:
```suggestion
def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
	def map = [name: "battery", unit: "%"]
	if (cmd.batteryLevel == 0xFF) {
		map.value = 1
		map.descriptionText = "$device.displayName has a low battery"
	} else {
		map.value = cmd.batteryLevel
	}
	state.lastbatt = new Date().time
	createEvent(map)
}
```

This is the battery handling code from zwave-siren. It would be good to be consistent. Specifically we turn 255 into 1%, not 0.

Reply:
you can ignore the 'lastbatt' stuff